pkg/service/cluster/service_integration_test.go

```diff
@@ -33,6 +33,114 @@ import (
 	"github.com/scylladb/scylla-manager/v3/pkg/util/uuid"
 )
 
+func TestValidateHostConnectivityIntegration(t *testing.T) {
+	// given
+	var (
+		ctx          = context.Background()
+		session      = CreateScyllaManagerDBSession(t)
+		secretsStore = store.NewTableStore(session, table.Secrets)
+		timeout      = 5 * time.Second
+		c            = &cluster.Cluster{
+			AuthToken: "token",
+			Host:      ManagedClusterHost(),
+		}
+	)
+	s, err := cluster.NewService(session, metrics.NewClusterMetrics(), secretsStore, scyllaclient.DefaultTimeoutConfig(),
+		server.DefaultConfig().ClientCacheTimeout, log.NewDevelopment())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = s.PutCluster(context.Background(), c)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	allHosts := ManagedClusterHosts()
+	for _, tc := range []struct {
+		name      string
+		hostsDown []string
+		result    error
+	}{
+		{
+			name:      "coordinator host is DOWN",
+			hostsDown: []string{ManagedClusterHost()},
+			result:    nil,
+		},
+		{
+			name:      "only one is UP",
+			hostsDown: allHosts[:len(allHosts)-1],
+			result:    nil,
+		},
+		{
+			name:      "all hosts are DOWN",
+			hostsDown: allHosts,
+			result:    cluster.ErrNoValidKnownHost,
+		},
+		{
+			name:      "all hosts are UP",
+			hostsDown: nil,
+			result:    nil,
+		},
+	} {
+		t.Run(tc.name, func(innerT *testing.T) {
+			defer func() {
+				for _, host := range tc.hostsDown {
+					if err := StartService(host, "scylla"); err != nil {
+						innerT.Logf("error on starting stopped scylla service on host={%s}, err={%s}", host, err)
+					}
+					if err := RunIptablesCommand(host, CmdUnblockScyllaREST); err != nil {
+						innerT.Logf("error trying to unblock REST API on host = {%s}, err={%s}", host, err)
+					}
```

Comment:
Being honest I don't know.
I want to reach the state where the API calls are hanging, so that the call may timeout. This part is done by calling the CmdBlockScyllaREST.
Besides that, I want the node to report DOWN status.

Let me do the check with just blocking the API.

Reply:
OK, so making the API unresponsive doesn't make the node to be DOWN.

When the scylla service is just stopped, then the VM (or container) refuses connection and there is no hanging on calling the API.

What makes sense, as Scylla is using gossip protocol to discover other nodes. Blocking gossip protocol would do the job I guess. But I don't see it as sommething necessary, as the service can be just stopped.

Reply:
Okay, so I get why blocking REST API is not enough, but wouldn't just stopping the node work?

Reply:
> When the scylla service is just stopped, then the VM (or container) refuses connection and there is no hanging on calling the API.

> What makes sense, as Scylla is using gossip protocol to discover other nodes. Blocking gossip protocol would do the job I guess. But I don't see it as sommething necessary, as the service can be just stopped.

The service must be stopped to block gossiper and cause the node to be reported as DOWN.

Reply:
I missed that part. Could you put it into comment, so that someone won't try to remove one of those parts in the future?

Reply:
Sure, let me put that into the comment